src/crypto/tls/metrics_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 Cloudflare, Inc. All rights reserved. Use of this source code
+// is governed by a BSD-style license that can be found in the LICENSE file.
+
+package tls
+
+import (
+	"crypto/x509"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"testing"
+	"time"
+)
+
+func (e EXP_EventTLS13ServerHandshakeTimingInfo) isMonotonic() bool {
```

Comment:
1. Suggestion: Instead of making this a method of of `EXP_EventTLS13Server...` I would just hard-code the checks in the test below. That way no one in the future stupidly tries to use this test-specific code elsewhere.
1. nit: `isMonotonicallyIncreasing` (here and below)
1. Hrm... this test would pass of all the durations were 0. I wonder if we should check that each duration is *strictly* greater than the last, and not just greater than or equal to the last?

Reply:
I went with a variation of this to keep the test code somewhat leaner.